src/Graph.elm

```diff
@@ -960,34 +961,42 @@ guidedDfs :
     -> ( acc, Graph n e )
 guidedDfs selectNeighbors visitNode startingSeeds startingAcc startingGraph =
     let
-        go seeds acc graph =
+        go : List NodeId -> Set NodeId -> acc -> Graph n e -> ( acc, Set NodeId, Graph n e )
+        go seeds visited acc graph =
```

Comment:
`graph` is a static argument; it is always `startingGraph`. No need to pass it around, pretending that it will change!